src/metadata/models/metadataManager.ts

```diff
@@ -0,0 +1,54 @@
+import { inject, injectable } from 'tsyringe';
+import { Repository } from 'typeorm';
+import { Services } from '../../common/constants';
+import { ILogger } from '../../common/interfaces';
+import { EntityNotFoundError, IdAlreadyExistsError } from './errors';
+import { IMetadata, Metadata, Payload } from './metadata';
+
+@injectable()
+export class MetadataManager {
+  public constructor(
+    @inject(Services.REPOSITORY) private readonly repository: Repository<Metadata>,
+    @inject(Services.LOGGER) private readonly logger: ILogger
+  ) {}
+
+  public async getAll(): Promise<IMetadata[] | undefined> {
+    this.logger.log('info', `Get all models metadata`);
+    return this.repository.find();
+  }
+
+  public async getRecord(identifier: string): Promise<IMetadata | undefined> {
+    this.logger.log('info', `Get metadata record ${identifier}`);
+    return this.repository.findOne(identifier);
+  }
+
+  public async createRecord(payload: Payload): Promise<IMetadata> {
+    this.logger.log('info', `Create a new metadata record: ${JSON.stringify(payload)}`);
+    const dbMetadata = await this.repository.findOne({ where: [{ identifier: payload.identifier }] });
+    if (dbMetadata != undefined) {
+      throw new IdAlreadyExistsError(`Metadata record ${dbMetadata.identifier} already exists`);
+    }
+    const links = payload.links?.map((link) => `,,${link.protocol},${link.url}`).join('^');
```

Comment:
The creation of links logic is present in both the create and update functions.
You should extract it into an helper function.

Reply:
if we already added the name and description fields to ILink we can insert them in the created link in case they exist. (name,description,protocol,url)

Reply:
Fixed